doc/classes/NavigationRegion2D.xml

```diff
@@ -10,6 +10,7 @@
 		The pathfinding cost of entering this region from another region can be controlled with the [member enter_cost] value.
 		[b]Note:[/b] This value is not added to the path cost when the start position is already inside this region.
 		The pathfinding cost of traveling distances inside this region can be controlled with the [member travel_cost] multiplier.
+		[b]Note:[/b] This node caches changes to its properties, so if you make changes to the underlying region [RID] in [NavigationServer2D], they will not be reflected in this node's properties.
 	</description>
```

Comment:
Just out of interest, why does the user need access to the underlying RID? Many Godot nodes use RIDs internally, but these are not generally exposed to the user, the user interacts using the client nodes and resources.

Reply:
This is a warning that if users use the NavigationServer API directly to change e.g. a navregion on a map or properties of the navregion then the NavigationRegion node will not show this changes cause it is displaying the cached values on the node. Maybe it could be worded different to make it more clear.

Half the NavigationServer API functions require RID's of the maps, regions, avoidance agents and depending on gameplay needs (e.g. multiple maps for actor sizes, procedural stuff, ai queries, ...) it can be a requirement to use the API to complement what is possible to do with SceneTree nodes alone.

Reply:
The question is not whether the warning is clear, but whether it is flagging a potential problem. In Godot the usual paradigm is as follows:

* Use nodes / resources - RIDs are hidden and unavailable to users
* Use RIDs and servers directly - do not use the nodes / resources

The mix and match approach is problematic, for exactly these types of reasons, which is why it is avoided. RIDs are essentially state machines operated by queue, and when you have two separate areas changing state independently it becomes much more difficult to prevent bugs. So the question is, what parts of the existing navigation system rely on a mix and match approach? Why are the RIDs exposed _at all_ from the nodes?

Reply:
I agree that it is a bit odd for nodes to be exposing their underlying RIDs, but I believe this is an issue best solved in another PR (assuming we come to an agreement).